scripts/o3de/ExportScripts/export_source_ios_xcode.py

```diff
@@ -0,0 +1,122 @@
+                             tools_build_folder: pathlib.Path,
+                             ios_build_folder: pathlib.Path,
+                             should_build_tools:bool = True,
+                             skip_asset_processing: bool = False,
+                             logger: logging.Logger|None = None):
+    """
+    This function serves as an initial exporter for project to the iOS platform. The steps in this code will generate
+    an Xcode project file containing all necessary build information to produce iOS executables. 
+    In order to build and deploy projects, it is recommended to open the project file in Xcode to configure and build accordingly.
+
+    Note: In order to use this functionality, you must be running this script from a MacOS machine with a valid copy of Xcode
+
+    Instructions to handle iOS projects in Xcode will be provided soon.
+
+    :param ctx:                                     The O3DE Script context provided by the export-command
+    :param tools_build_folder:                      Optional build path to build the tools. (Will default to build/tools if not supplied)
+    :param ios_build_folder:                        The base output path of the generated Xcode Project file for iOS
+    :param should_build_tools:                      Option to build the export process dependent tools (AssetProcessor, AssetBundlerBatch, and dependencies)
+    :param should_build_all_assets:                 Option to process all the assets for the game
+    :param logger:                                  Optional logger to use to log the process and errors
+    """
+    
+    # Use a provided logger or get the current system one
+    if not logger:
+        logger = logging.getLogger()
+        logger.setLevel(logging.ERROR)
+
+    tools_build_folder_str = str(tools_build_folder)
+    ios_build_folder_str = str(ios_build_folder)
+
+    # Optionally build the toolchain needed to process the assets
+    if should_build_tools:
+        process_command(["cmake", "-B", tools_build_folder_str, '-G', "Xcode","-DLY_UNITY_BUILD=ON"], cwd=ctx.project_path)
+
+        process_command(["cmake", "--build", tools_build_folder_str, "--target", "Editor", "AssetProcessorBatch", "--config", "profile"],
+                    cwd=ctx.project_path)
+        
+    # Optionally process the assets
+    if not skip_asset_processing:
+        asset_processor_batch_path = exp.get_asset_processor_batch_path(tools_build_folder, True)
+        process_command([ str(asset_processor_batch_path), '--platforms=ios' ,
+                        '--project-path', ctx.project_path ], cwd=ctx.project_path)
+
+    # Generate the Xcode project file for the O3DE project
+    cmake_toolchain_path = ctx.engine_path / 'cmake/Platform/iOS/Toolchain_ios.cmake'
+
+    process_command(['cmake', '-B', ios_build_folder_str, '-G', "Xcode", f'-DCMAKE_TOOLCHAIN_FILE={str(cmake_toolchain_path)}', '-DLY_UNITY_BUILD=ON', '-DLY_MONOLITHIC_GAME=1'],
```

Comment:
repeat: `LY_UNITY_BUILD=ON` isn't needed since it is the default value
```suggestion
    process_command(['cmake', '-B', ios_build_folder_str, '-G', "Xcode", f'-DCMAKE_TOOLCHAIN_FILE={str(cmake_toolchain_path)}', '-DLY_MONOLITHIC_GAME=1'],
```